openwisp_monitoring/device/api/views.py

```diff
@@ -203,6 +205,72 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        if 'resources' not in data:
+            return
+        self._write_cpu(data['resources'], pk, ct)
+        self._write_disk(data['resources'], pk, ct)
+        self._write_memory(data['resources'], pk, ct)
+
+    def _write_cpu(self, resources, primary_key, content_type):
+        extra_values = {
+            'load_1': float(resources['load'][0]),
+            'load_5': float(resources['load'][1]),
+            'load_15': float(resources['load'][2]),
+        }
+        metric, created = Metric._get_or_create(
+            object_id=primary_key,
+            content_type=content_type,
+            key='cpu',
+            field_name='cpu_usage',
+            name='Resources: CPU',
+        )
+        metric.write(
+            float(resources['load'][0] / resources['cpus']), extra_values=extra_values,
+        )
+        if created:
+            self._create_resources_graph(metric, resource='cpu')
+            self._create_resources_threshold(metric, resource='cpu')
+
+    def _write_disk(self, resources, primary_key, content_type):
+        used_bytes, size_bytes, available_bytes = 0, 0, 0
+        for disk in resources['disk']:
+            used_bytes += disk['used_bytes']
+            size_bytes += disk['size_bytes']
+            available_bytes += disk['available_bytes']
+        metric, created = Metric._get_or_create(
+            object_id=primary_key,
+            content_type=content_type,
+            key='disk',
+            field_name='used_disk',
+            name='Resources: Disk',
+        )
+        metric.write(used_bytes / size_bytes)
+        if created:
+            self._create_resources_graph(metric, resource='disk')
+            self._create_resources_threshold(metric, resource='disk')
+
+    def _write_memory(self, resources, primary_key, content_type):
+        memory = resources['memory']
+        extra_values = {
+            'total_memory': memory['total'],
+            'free_memory': memory['free'],
+            'buffered_memory': memory['buffered'],
+            'shared_memory': memory['shared'],
+            'available_memory': memory['available'],
+            'cached_memory': memory['cached'],
+        }
+        metric, created = Metric._get_or_create(
+            object_id=primary_key,
+            content_type=content_type,
+            key='memory',
+            field_name='used_memory',
+            name='Resources: Memory',
+        )
+        used_memory = 1 - (memory['free'] + memory['buffered']) / memory['total']
```

Comment:
as we discussed, here we need is to:

- consider `available` optional (we need a test for this, which tests the metric collection without available, because some systems don't have it and older openwrt versions don't have it)
- if `available` is present, use it instead of `free`, but only if it's greater than `free`